src/ddimm_parser.cpp

```diff
@@ -0,0 +1,263 @@
+                                     constants::VALUE_3) +
+                                    (iterator[constants::SPD_BYTE_234] &
+                                     constants::MASK_BYTE_BITS_012) +
+                                    constants::VALUE_2;
+
+        if (!checkValidValue(iterator[constants::SPD_BYTE_6] &
+                                 constants::MASK_BYTE_BITS_567,
+                             constants::SHIFT_BITS_5, constants::VALUE_0,
+                             constants::VALUE_3))
+        {
+            logging::logMessage(
+                "Capacity calculation failed for dram width DDIMM Byte 6 value "
+                "[" +
+                std::to_string(iterator[constants::SPD_BYTE_6]) + "]");
+            break;
+        }
+        uint8_t l_dramWidth =
+            constants::VALUE_4 *
+            (constants::VALUE_1 << ((iterator[constants::SPD_BYTE_6] &
+                                     constants::MASK_BYTE_BITS_567) >>
+                                    constants::VALUE_5));
+
+        dimmSize = (l_channelsPerDdimm * l_busWidthPerChannel *
+                    l_diePerPackage * l_densityPerDie * l_ranksPerChannel) /
+                   (8 * l_dramWidth);
+
+    } while (false);
+
+    return constants::CONVERT_MB_TO_KB * dimmSize;
+}
+
+size_t
+    DdimmVpdParser::getDdimmSize(types::BinaryVector::const_iterator iterator)
+{
+    size_t dimmSize = 0;
+    if ((iterator[constants::SPD_BYTE_2] & constants::SPD_BYTE_MASK) ==
+        constants::SPD_DRAM_TYPE_DDR5)
+    {
+        dimmSize = getDdr5BasedDdimmSize(iterator);
+    }
+    else
+    {
+        logging::logMessage("Error: DDIMM is not DDR5. DDIMM Byte 2 value [" +
+                            std::to_string(iterator[constants::SPD_BYTE_2]) +
+                            "]");
+    }
+    return dimmSize;
+}
+
+void DdimmVpdParser::readKeywords(types::BinaryVector::const_iterator iterator)
+{
+    // collect Dimm size value
+    auto dimmSize = getDdimmSize(iterator);
+    if (!dimmSize)
+    {
+        throw(DataException("Error: Calculated dimm size is 0."));
+    }
+
+    m_parsedVPDMap.emplace("MemorySizeInKB", dimmSize);
+    // point the iterator to DIMM data and skip "11S"
+    advance(iterator, constants::MEMORY_VPD_DATA_START + 3);
+    types::BinaryVector partNumber(iterator,
+                                   iterator + constants::PART_NUM_LEN);
+
+    advance(iterator, constants::PART_NUM_LEN);
+    types::BinaryVector serialNumber(iterator,
+                                     iterator + constants::SERIAL_NUM_LEN);
+
+    advance(iterator, constants::SERIAL_NUM_LEN);
+    types::BinaryVector ccin(iterator, iterator + constants::CCIN_LEN);
+
+    m_parsedVPDMap.emplace("FN", partNumber);
+    m_parsedVPDMap.emplace("PN", move(partNumber));
+    m_parsedVPDMap.emplace("SN", move(serialNumber));
+    m_parsedVPDMap.emplace("CC", move(ccin));
+}
+
+types::VPDMapVariant DdimmVpdParser::parse()
```

Comment:
why is VPD in caps for VPDMapVariant?

Reply:
I have created issue for this https://github.com/ibm-openbmc/openpower-vpd-parser/issues/196.
Will update the code changes while resolving the above mentioned issue.